### ISSUE TITLE
targetSdkVersionFromManifest fails if apk does not include targetSdkVersion

### ISSUE THREAD

**Issue by rkavalap** [appium/appium#7394]
@john7doe commented on [Tue Dec 06 2016](https://github.com/appium/appium-adb/issues/188)

Steps to reproduce:

copy settings_apk-debug.apk from io.appium.settings to test/fixtures/settings_apk-debug.apk

change the following in android-manifest-e2e-specs.js:
```
-                                'selendroid', 'selendroid.apk');
+                                'selendroid', 'selendroid.apk'),
+      settingsApp = path.resolve(rootDir, 'test', 'fixtures', 'settings_apk-debug.apk');
```

add this to the 'Android-manifest' spec:

 it('targetSdkVersionFromManifest works on our own settingsApp', async () => {
    let version = adb.targetSdkVersionFromManifest(settingsApp);
});

---

@john7doe commented on [Tue Dec 06 2016](https://github.com/appium/appium-adb/issues/188#issuecomment-265111934)

It could be that this behaviour is intended, but in that case then the use in apk-utils.js need to take this into account in its install method. Its currently preventing android testing on v1.6.2 as the settings_apk-debug.apk does not have a targetSdkVersion in its manifest

---

@john7doe commented on [Tue Dec 06 2016](https://github.com/appium/appium-adb/issues/188#issuecomment-265132078)

related to https://github.com/appium/appium/issues/7349

**Comment by filmaj** [comment]
+1, I have also hit this :(

**Comment by jlipps** [comment]
issue's fixed in appium-adb so will be resolved once we publish appium 1.6.3
